binnacle/lib/providers/navigation_provider.dart

```diff
@@ -1,65 +1,284 @@
 import 'dart:async';
-
 import 'package:flutter/foundation.dart';
 import 'package:google_maps_flutter/google_maps_flutter.dart';
 import 'package:rxdart/rxdart.dart';
+import 'dart:math';
+import 'package:vector_math/vector_math.dart';
 import 'package:sail_routing_dart/route_model.dart';
-import 'package:sos/enums.dart';
+import 'package:sail_routing_dart/polar_plotting/polar_plot.dart';
+import 'package:sail_routing_dart/polar_algs/polar_router.dart';
+import 'package:sail_routing_dart/shared/route_math.dart';
 import 'package:sos/models/position_model.dart';
-import 'package:vector_math/vector_math.dart';
+import 'package:sos/models/ideal_heading_model.dart';
+import 'package:sos/models/wind_model.dart';
+import 'package:sos/enums.dart';
+import 'package:flutter/services.dart' show rootBundle;
 
+const metersPerDegree = 111111;
+const defaultMaxOffset = 25.0;
+const defaultCloseEnough = 25.0;
+/// Navigation Provider handles new positions and route calculations.
+/// Every position change checks if a new event needs to be emitted or if the route needs to be recalculated.
```

Comment:
I think this should be expanded to include explaining how it works along a route and knows which checkpoint it is trying to get to now.

Reply:
Documentation of these are explained in maxOffset and closeEnough public getters and setters now.